examples/mqtt-paho/main.c

```diff
@@ -0,0 +1,191 @@
+#include "net/ipv4/addr.h"
+#include "MQTTClient.h"
+#include "shell.h"
+
+#define REMOTE_IP_ADDRESS       "fe80::18cf:f9ff:fe1c:afd2"
+#define MAX_MSGS                12
+#define BUF_SIZE                250
+
+static char* topic = "test_from_riot";
+MQTTClient client;
+static char _sender_stack[THREAD_STACKSIZE_DEFAULT];
+
+static void _messageArrived(MessageData* data)
+{
+    printf("mqtt_example: message arrived on topic"
+          " %.*s: %.*s\n\n", (int)data->topicName->lenstring.len, data->topicName->lenstring.data,
+      (int)data->message->payloadlen, (char *)data->message->payload);
+}
+
+static int ifconfig(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    for (struct netif *iface = netif_list; iface != NULL; iface = iface->next) {
+        printf("%s_%02u: ", iface->name, iface->num);
+#ifdef MODULE_LWIP_IPV6
+        char addrstr[IPV6_ADDR_MAX_STR_LEN];
+        for (int i = 0; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
+            if (!ipv6_addr_is_unspecified((ipv6_addr_t *)&iface->ip6_addr[i])) {
+                printf(" inet6 %s\n", ipv6_addr_to_str(addrstr, (ipv6_addr_t *)&iface->ip6_addr[i],
+                                                       sizeof(addrstr)));
+            }
+        }
+#endif
+#ifdef MODULE_LWIP_IPV4
+        char addrstr[IPV4_ADDR_MAX_STR_LEN];
+        printf(" inet %s\n", ipv4_addr_to_str(addrstr, (ipv4_addr_t *)&iface->ip_addr,
+                                              sizeof(addrstr)));
+#endif
+        puts("");
+    }
+    return 0;
+}
+
+static void *_mqtt_sender(void *arg)
+{
+    (void) arg;
+    int count = 1;
+    for (;;) {
+        MQTTMessage message;
+        char payload[30];
+
+        message.qos = 1;
+        message.retained = 0;
+        message.payload = payload;
+        sprintf(payload, "message number %d", count);
+        message.payloadlen = strlen(payload);
+
+        int rc;
+        if ((rc = MQTTPublish(&client, topic, &message)) != 0) {
+            printf("mqtt_example: return code from MQTT publish is %d\n", rc);
+        } else {
+            printf("Message (%s) has been published to topic:%s\n", (char *)message.payload, topic);
+        }
+        if (count >= MAX_MSGS) {
+            /* kill thread */
+            printf("mqtt_example: finishing example, check that %d messages were sent/received\n",
+                     count);
+            return NULL;
+        }
+        count++;
+    }
+    return NULL;
+}
+
+static void usage(void)
+{
+    puts("Usage:");
+    puts("  help: this help");
+    puts("  pub msg: publish 'msg' to some random 'topic'");
+}
+
+static int mqtt_publish_cmd(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+
+    if (argc < 2) {
+        usage();
+        return 1;
+    }
+
+#define STR_LEN     25
+    MQTTMessage msg;
+    msg.qos = QOS1;
+    if (strcmp(argv[0], "pub") == 0) {
+        msg.payload = argv[1];
+        msg.payloadlen = strlen(argv[1]);
+        MQTTPublish(&client, topic, &msg);
```

Comment:
It will be more convenient to be able to use a custom MQTT topic retrieved from the command line instead of the hard coded one provided at the beginning of the file.